exchanges/populate/src/populateExchange.ts

```diff
@@ -84,6 +183,8 @@ export const populateExchange = ({
     }
 
     parsedOperations.add(key);
+    currentVariables = variables || {};
+    readFromQuery(query);
 
     const [extractedFragments, newFragments] = extractSelectionsFromQuery(
```

Comment:
we should remove this line as

> Fragments won't be tracked anymore. It's assumed that only missing fields from a given selection set are added and that it doesn't matter how they're added, since the API does not care.